Combine aliens and shorten methods

harveyvanveltom opened the pull request.

harveyvanveltom committed:
<features,logic combined alienfactory with the shorten methods code

harveyvanveltom committed:
<debug fixed the score

now when using multiple aliens at the same time each kind of alien keeps
its own score.

harveyvanveltom committed:
<testing,debug fixed testing + added new alien tests

Mari3 commented:
Why is the aliens vector passed as a parameter for the createAliens method? I think that's not neccesary because that vector is an attribute of the game class

harveyvanveltom committed:
<logic removed unnecessary parsing

harveyvanveltom commented:
you were right. I now removed the parameter from the createAliens method.

JerPk commented on src/main/java/spaceinvaders/spaceinvaders_framework/AlienType2.java:
```diff
@@ -0,0 +1,44 @@
+    public Bullet shoot() {
+      final SpriteSheet spritesheet = new SpriteSheet(getGame().getSpriteSheet());
+      final Bullet newBullet = new Bullet(getX() + 5, getY() + 2, spritesheet);
+      Game.logfile.writeShoot("Alien", getX(), getY());
```
Maybe you can add the type of alien to the logfile.

JerPk commented on src/main/java/spaceinvaders/spaceinvaders_framework/AlienType3.java:
```diff
@@ -0,0 +1,44 @@
+    public Bullet shoot() {
+      final SpriteSheet spritesheet = new SpriteSheet(getGame().getSpriteSheet());
+      final Bullet newBullet = new Bullet(getX() + 5, getY() + 2, spritesheet);
+      Game.logfile.writeShoot("Alien", getX(), getY());
```
Same as in Alientype2.java (add type of alien)

Jeroemaster commented:
Can be merged after the changes for the logfile

harveyvanveltom committed:
<feature specified logging aliens.

alien constructor and alien bullets logging now specify which alientype
performs is created and which type shoots the bullet.
